com.unity.multiplayer.mlapi/Tests/Runtime/NetworkPrefabHandlerTests.cs

```diff
@@ -0,0 +1,162 @@
+
+            //Register via NetworkObject
+            gameObjectRegistered = networkPrefabHandler.AddHandler(baseObject, networkPrefaInstanceHandler);
+
+            //Test result of registering via NetworkObject reference
+            Assert.True(gameObjectRegistered);
+
+            //Change it up
+            prefabPosition = new Vector3(2.0f, 1.0f, 5.0f);
+            prefabRotation = new Quaternion(4.0f, 1.5f, 5.4f, 5.1f);
+
+            spawnedObject = networkPrefabHandler.HandleNetworkPrefabSpawn(baseObject.GlobalObjectIdHash, 0, prefabPosition, prefabRotation);
+
+            //Test that something was instantiated
+            Assert.NotNull(spawnedObject);
+
+            //Test that this is indeed an instance of our original object
+            Assert.True(spawnedObject.name.Contains("NetworkPrefabHandlerTestObject"));
+
+            //Test for position and rotation
+            Assert.True(prefabPosition == spawnedObject.transform.position);
+            Assert.True(prefabRotation == spawnedObject.transform.rotation);
+
+            networkPrefabHandler.HandleNetworkPrefabDestroy(spawnedObject);     //Destroy our prefab instance
+            networkPrefabHandler.RemoveHandler(baseObject);                     //Remove our handler
+
+            //Register via GlobalObjectHashId
+            gameObjectRegistered = networkPrefabHandler.AddHandler(baseObject.GlobalObjectIdHash, networkPrefaInstanceHandler);
+
+            //Test result of registering via GlobalObjectHashId reference
+            Assert.True(gameObjectRegistered);
+
+            //Change it up
+            prefabPosition = new Vector3(6.0f, 4.0f,1.0f);
+            prefabRotation = new Quaternion(3f, 2f, 4f, 1f);
+
+            spawnedObject = networkPrefabHandler.HandleNetworkPrefabSpawn(baseObject.GlobalObjectIdHash, 0, prefabPosition, prefabRotation);
+
+            //Test that something was instantiated
+            Assert.NotNull(spawnedObject);
+
+            //Test that this is indeed an instance of our original object
+            Assert.True(spawnedObject.name.Contains("NetworkPrefabHandlerTestObject"));
+
+            //Test for position and rotation
+            Assert.True(prefabPosition == spawnedObject.transform.position);
+            Assert.True(prefabRotation == spawnedObject.transform.rotation);
+
+            networkPrefabHandler.HandleNetworkPrefabDestroy(spawnedObject);     //Destroy our prefab instance
+            networkPrefabHandler.RemoveHandler(baseObject);                     //Remove our handler
+
+            Assert.False(networkPrefaInstanceHandler.StillHasInstances());
+        }
+
+        public void Dispose()
+        {
+            //Stop, shutdown, and destroy
+            NetworkManagerHelper.ShutdownNetworkManager();
+        }
+
+        public NetworkPrefabHandlerTests()
+        {
+            //Create, instantiate, and host
+            NetworkManagerHelper.StartNetworkManager();
+        }
```

Comment:
you could also use `[SetUp]` and `[TearDown]` attributed methods instead of `class ctor` and `Dispose()`
- https://docs.unity3d.com/Packages/com.unity.test-framework@1.1/manual/reference-actions-outside-tests.html
- https://docs.nunit.org/articles/nunit/technical-notes/usage/SetUp-and-TearDown.html

Reply:
SetUpAttribute **is now used exclusively for per-test setup**.
TearDownAttribute **is now used exclusively for per-test teardown**.

The NetworkManagerHelper method for unit tests **is meant to stay active during the duration of all tests selected** to be run.  While this is one test, there are many other tests that could be run that might need to use the NetworkManager instance created for the temporary test scene and I also plan on adding more tests to NetworkPrefabHandlerTests.

The reason for the change in the setup and tear down is to deal with scenarios where you allocate something but have an assert that breaks the current stack scope (_i.e. if you tried to allocate something that needs to be disposed/destroyed during a test and expected to do these actions at the end of your test but triggered an Assert...you would never dispose/destroy said allocated objects...thus a per test setup and tear down pattern_)

Reply:
Replaced the constructor and dispose methods with Setup and TearDown attributes.
Made changes to NetworkManagerHelper to handle the remaining issues with instantiating and destroying NetworkManager multiple times.